File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/character/alignment/ActorExtendedAlignmentInformations.scala





















// Generated on 05/03/2013 10:35:02
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait ActorExtendedAlignmentInformations extends ActorAlignmentInformations {
  override val typeId = 202

  protected trait Type extends ActorAlignmentInformations.Type with Serializable {
    override val typeId = ActorExtendedAlignmentInformations.typeId

    val honor: UShort
    val honorGradeFloor: UShort
    val honorNextGradeFloor: UShort
    val pvpEnabled: Boolean

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeUShort(honor)
      buf.writeUShort(honorGradeFloor)
      buf.writeUShort(honorNextGradeFloor)
      buf.writeBoolean(pvpEnabled)
    }

  }

  protected def create(alignmentSide: Byte, alignmentValue: Byte, alignmentGrade: Byte, dishonor: UShort, characterPower: Int, honor: UShort, honorGradeFloor: UShort, honorNextGradeFloor: UShort, pvpEnabled: Boolean): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val alignmentSide = buf.readByte()
    val alignmentValue = buf.readByte()
    val alignmentGrade = buf.readByte()
    val dishonor = buf.readUShort()
    val characterPower = buf.readInt()
    val honor = buf.readUShort()
    val honorGradeFloor = buf.readUShort()
    val honorNextGradeFloor = buf.readUShort()
    val pvpEnabled = buf.readBoolean()

    create(alignmentSide, alignmentValue, alignmentGrade, dishonor, characterPower, honor, honorGradeFloor, honorNextGradeFloor, pvpEnabled)
  }

}

object ActorExtendedAlignmentInformations extends ActorExtendedAlignmentInformations {
  case class TypeImpl(alignmentSide: Byte, alignmentValue: Byte, alignmentGrade: Byte, dishonor: UShort, characterPower: Int, honor: UShort, honorGradeFloor: UShort, honorNextGradeFloor: UShort, pvpEnabled: Boolean) extends Type
  override def create(alignmentSide: Byte, alignmentValue: Byte, alignmentGrade: Byte, dishonor: UShort, characterPower: Int, honor: UShort, honorGradeFloor: UShort, honorNextGradeFloor: UShort, pvpEnabled: Boolean) = TypeImpl(alignmentSide, alignmentValue, alignmentGrade, dishonor, characterPower, honor, honorGradeFloor, honorNextGradeFloor, pvpEnabled)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(alignmentSide: Byte, alignmentValue: Byte, alignmentGrade: Byte, dishonor: UShort, characterPower: Int, honor: UShort, honorGradeFloor: UShort, honorNextGradeFloor: UShort, pvpEnabled: Boolean) = create(alignmentSide, alignmentValue, alignmentGrade, dishonor, characterPower, honor, honorGradeFloor, honorNextGradeFloor, pvpEnabled)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/ActorExtendedAlignmentInformations.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ActorExtendedAlignmentInformations extends ActorAlignmentInformations {
    public static final short TYPE_ID = 202;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int honor;
    public int honorGradeFloor;
    public int honorNextGradeFloor;
    public boolean pvpEnabled;
    
    public ActorExtendedAlignmentInformations() { }
    
    public ActorExtendedAlignmentInformations(byte alignmentSide, byte alignmentValue, byte alignmentGrade, int dishonor, int characterPower, int honor, int honorGradeFloor, int honorNextGradeFloor, boolean pvpEnabled) {
        super(alignmentSide, alignmentValue, alignmentGrade, dishonor, characterPower);
        this.honor = honor;
        this.honorGradeFloor = honorGradeFloor;
        this.honorNextGradeFloor = honorNextGradeFloor;
        this.pvpEnabled = pvpEnabled;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeUShort(honor);
        buf.writeUShort(honorGradeFloor);
        buf.writeUShort(honorNextGradeFloor);
        buf.writeBoolean(pvpEnabled);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        honor = buf.readUShort();
        if (honor < 0 || honor > 20000)
            throw new RuntimeException("Forbidden value on honor = " + honor + ", it doesn't respect the following condition : honor < 0 || honor > 20000");
        honorGradeFloor = buf.readUShort();
        if (honorGradeFloor < 0 || honorGradeFloor > 20000)
            throw new RuntimeException("Forbidden value on honorGradeFloor = " + honorGradeFloor + ", it doesn't respect the following condition : honorGradeFloor < 0 || honorGradeFloor > 20000");
        honorNextGradeFloor = buf.readUShort();
        if (honorNextGradeFloor < 0 || honorNextGradeFloor > 20000)
            throw new RuntimeException("Forbidden value on honorNextGradeFloor = " + honorNextGradeFloor + ", it doesn't respect the following condition : honorNextGradeFloor < 0 || honorNextGradeFloor > 20000");
        pvpEnabled = buf.readBoolean();
    }
    
}
